cli/src/cli.rs

```diff
@@ -1131,24 +1146,65 @@ fn process_deploy(
         bpf_loader::id()
     };
 
-    // Build transactions to calculate fees
-    let mut messages: Vec<&Message> = Vec::new();
-    let (blockhash, fee_calculator, _) = rpc_client
-        .get_recent_blockhash_with_commitment(config.commitment)?
-        .value;
     let minimum_balance = rpc_client.get_minimum_balance_for_rent_exemption(program_data.len())?;
-    let ix = system_instruction::create_account(
-        &config.signers[0].pubkey(),
-        &program_id.pubkey(),
-        minimum_balance.max(1),
-        program_data.len() as u64,
-        &loader_id,
-    );
-    let message = Message::new(&[ix], Some(&config.signers[0].pubkey()));
-    let mut create_account_tx = Transaction::new_unsigned(message);
     let signers = [config.signers[0], program_id];
-    create_account_tx.try_sign(&signers, blockhash)?;
-    messages.push(&create_account_tx.message);
+
+    // Check program account to see if partial initialization has occurred
+    println!("before account check");
+    let initial_instructions = if let Some(account) = rpc_client
+        .get_account_with_commitment(&program_id.pubkey(), config.commitment)?
+        .value
+    {
```

Comment:
It seems like this assumes either we either get a new account or the failed account.  Should we also check for the account being occupied otherwise and bail?

Reply:
I like that idea. We may want to update web3.js with some of these additional checks.